src/main/java/seedu/duke/task/Task.java

```diff
@@ -0,0 +1,84 @@
+package seedu.duke.task;
+
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Used to create task that is added to {@link TaskList}.
+ */
+public abstract class Task {
+    /**
+     * taskName refers to the name of the task.
+     */
+    private final String taskName;
+    /**
+     * done records if a task has been marked as complete.
+     */
+    private final boolean done;
+    private final LocalDateTime date;
+
+    /**
+     * Constructor for a task.
+     * @param name refers to the task name
+     */
+    Task(String name) {
+        this.taskName = name;
+        this.done = false;
```

Comment:
Perhaps consider naming it to be more boolean sounding, like `isDone`!

Reply:
renamed! Thanks for spotting this